packages/widgets/src/docker/component.tsx

```diff
@@ -0,0 +1,311 @@
+                size="var(--button-size)"
+                onClick={() => handleActionAsync("restart")}
+              >
+                <IconRotateClockwise style={actionIconIconStyle} />
+              </ActionIcon>
+            </Tooltip>
+          </Group>
+        );
+      },
+    },
+  ];
+
+  const { data } = clientApi.docker.getContainersWidget.useQuery();
+  const relativeTime = useTimeAgo(data?.timestamp ? new Date(data.timestamp) : new Date());
+
+  const totalContainers = data?.data.containers.length ?? 0;
+  const containerStateCounts = data?.data.containers.reduce<Record<ContainerState, number>>(
+    (acc, container) => {
+      acc[container.state as ContainerState] = (acc[container.state as ContainerState] || 0) + 1;
+      return acc;
+    },
+    {
+      created: 0,
+      running: 0,
+      paused: 0,
+      restarting: 0,
+      exited: 0,
+      removing: 0,
+      dead: 0,
+    },
+  );
+
+  const baseStyle: MantineStyleProp = {
+    "--total-width": "calc(100cqw / var(--total-width))",
+    "--ratio-width": "calc(100cqw / var(--total-width))",
+    "--space-size": "calc(var(--ratio-width) * 0.1)", //Standard gap and spacing value
+    "--text-fz": "calc(var(--ratio-width) * 0.45)", //General Font Size
+    "--icon-size": "calc(var(--ratio-width) * 2 / 3)", //Normal icon size
+    "--mrt-base-background-color": "transparent",
+  };
+
+  const table = useTranslatedMantineReactTable({
+    columns: columns(),
+    data: data?.data.containers ?? [],
+    enablePagination: false,
+    enableTopToolbar: false,
+    enableBottomToolbar: false,
+    enableSorting: false,
+    enableColumnActions: false,
+    enableStickyHeader: false,
+    enableColumnOrdering: false,
+    enableRowSelection: false,
+    enableFullScreenToggle: false,
+    enableGlobalFilter: false,
+    enableDensityToggle: false,
+    enableFilters: false,
+    enableHiding: false,
+    initialState: {
+      density: "xs",
+    },
+    mantinePaperProps: {
+      flex: 1,
+      withBorder: false,
+      shadow: undefined,
+    },
+    mantineTableProps: {
+      className: "docker-widget-table",
+      style: {
+        tableLayout: "fixed",
+      },
+    },
+    mantineTableContainerProps: {
+      style: {
+        height: "100%",
+      },
+    },
+  });
+
+  return (
+    <Stack gap={0} h="100%" display="flex" style={baseStyle}>
+      <MantineReactTable table={table} />
+      <Group
+        gap="1cqmin"
+        h="var(--ratio-width)"
+        px="var(--space-size)"
+        pr="3cqmin"
+        pl="3cqmin"
+        justify="space-between"
+        style={{
+          borderTop: "0.0625rem solid var(--border-color)",
+        }}
+      >
+        <Group gap={1}>
+          <IconBrandDocker size="8cqmin" />
+          <Text size="6cqmin">{t("table.footer", { count: totalContainers })}</Text>
+        </Group>
+        <Group gap="2cqmin">
+          {containerStateCounts &&
+            Object.entries(containerStateCounts).map(([state, count]) =>
```

Comment:
Use typesafe `objectEntries` from `@homarr/common` to no longer need the `as keyof typeof containerStates`